src/test/java/uk/gov/pay/ledger/util/pagination/PaginationBuilderTest.java

```diff
@@ -138,4 +141,27 @@ public void shouldNotGenerateNextLink_whenLimitTotalIsSetAndCountIsLessThanPageS
         builder = builder.buildResponse();
         assertThat(builder.getNextLink(), is(nullValue()));
     }
-}
\ No newline at end of file
+    
+    @Test
+    public void shouldGenerateSelfLinkWithOnlyPrescribedSpecialCharactersURLEncoded(){
+        Map<String, Object> valuesForReferenceParameter = Map.of(
+                "{ref", "reference=%7Bref",
+                "[ref", "reference=%5Bref",
+                "r{e{f}{", "reference=r%7Be%7Bf%7D%7B",
+                "ref=", "reference=ref=",
+                "ref&", "reference=ref&",
+                "ref@", "reference=ref%40"
+        );
+        for (Map.Entry<String, Object> entry : valuesForReferenceParameter.entrySet()) {
```

Comment:
We should use [parameterised tests](https://junit.org/junit5/docs/current/user-guide/#writing-tests-parameterized-tests) to check multiple inputs

Reply:
Only just noticed this in the [java coding standards](https://pay-team-manual.cloudapps.digital/manual/coding-standards/java-coding-standards.html) manual page 🤦‍♂️. I'll redo this.